astropy/coordinates/tests/test_finite_difference_velocities.py

```diff
@@ -0,0 +1,172 @@
+    suni = ICRS(ra=msungr.lon, dec=msungr.lat, distance=100*u.au,
+                pm_ra_cosdec=0*u.marcsec/u.yr, pm_dec=0*u.marcsec/u.yr,
+                radial_velocity=0*u.km/u.s)
+    qtrsuni = ICRS(ra=qtrsung.ra, dec=qtrsung.dec, distance=100*u.au,
+                   pm_ra_cosdec=0*u.marcsec/u.yr, pm_dec=0*u.marcsec/u.yr,
+                   radial_velocity=0*u.km/u.s)
+    sung = suni.transform_to(gf)
+    qtrsung = qtrsuni.transform_to(gf)
+
+    # should be high along the ecliptic-not-sun sun axis and
+    # low along the sun axis
+    assert np.abs(qtrsung.radial_velocity) > 30*u.km/u.s
+    assert np.abs(qtrsung.radial_velocity) < 40*u.km/u.s
+    assert np.abs(sung.radial_velocity) < 1*u.km/u.s
+
+    suni2 = sung.transform_to(ICRS)
+    assert np.all(np.abs(suni2.data.differentials['s'].d_xyz) < 3e-5*u.km/u.s)
+    qtrisun2 = qtrsung.transform_to(ICRS)
+    assert np.all(np.abs(qtrisun2.data.differentials['s'].d_xyz) < 3e-5*u.km/u.s)
+
+
+def test_altaz_diffs():
+    time = Time('J2015') + np.linspace(-1, 1, 1000)*u.day
+    loc = EarthLocation.of_site('greenwich')  #built-in
+    aa = AltAz(obstime=time, location=loc)
+
+    icoo = ICRS(np.zeros_like(time)*u.deg, 10*u.deg, 100*u.au,
+         pm_ra_cosdec=np.zeros_like(time)*u.marcsec/u.yr, pm_dec=0*u.marcsec/u.yr,
+         radial_velocity=0*u.km/u.s)
+
+    acoo = icoo.transform_to(aa)
+
+    # make sure the change in radial velocity over ~2 days isn't much more than
+    # the rotation speed of the Earth
+    assert np.ptp(acoo.radial_velocity)/2 < (2*np.pi*constants.R_earth/u.day)*1.2
+
+    cdiff = acoo.data.differentials['s'].represent_as(CartesianDifferential,
+                                                    acoo.data)
+
+    # The "total" velocity should be > c, because the *tangential* velocity
+    # isn't a True velocity, but rather an induced velocity due to the Earth's
+    # rotation at a distance of 100 AU
+    assert np.all(np.sum(cdiff.d_xyz**2, axis=0)**0.5 > constants.c)
+
+_xfail = pytest.mark.xfail
+@pytest.mark.parametrize('distance', [1000*u.au,
+                                      10*u.pc,
+                                      pytest.param(10*u.kpc, marks=_xfail),
+                                      pytest.param(100*u.kpc, marks=_xfail)])
+def test_numerical_limits(distance):
+    """
+    Tests the numerical stability of the default settings for the finite
+    difference transformation calculation.  This is *known* to fail for at
+    >~1kpc, but this may be improved in future versions.
+    """
+    time = Time('J2017') + np.linspace(-.5, .5, 100)*u.year
+
+    icoo = ICRS(ra=0*u.deg, dec=10*u.deg, distance=distance,
+                pm_ra_cosdec=0*u.marcsec/u.yr, pm_dec=0*u.marcsec/u.yr,
+                radial_velocity=0*u.km/u.s)
+    gcoo = icoo.transform_to(GCRS(obstime=time))
+    rv =  gcoo.radial_velocity.to('km/s')
+
+    # if its a lot bigger than this, finite-difference noise has ruined things
+    assert np.ptp(rv) < 65*u.km/u.s
```

Comment:
Not quite - it's empirically determined based on that direction and distances where the finite-difference issues are not present - so its more geometric (will add a comment)